app/src/main/java/ch/epfl/sweng/rps/ui/home/MatchViewModel.kt

```diff
@@ -20,51 +22,62 @@ import java.util.*
 class MatchViewModel : ViewModel() {
 
 
-    private var _gameService: OfflineGameService? = null
+    var gameService: GameService? = null
     var currentUserResult: Hand.Result? = null
+    var computerPlayer: ComputerPlayer? = null
     private val uid = FirebaseAuth.getInstance().currentUser?.uid
     fun startOfflineGameService(
         nEvents: Int,
         computerPlayer: ComputerPlayer,
         artificialMovesDelay: Long = 1_000
     ) {
         // TODO: when having a local DB change the random UUID
+        this.computerPlayer = computerPlayer
         val gameId = UUID.randomUUID().toString()
-        _gameService = OfflineGameService(
+        gameService = OfflineGameService(
             gameId,
             ServiceLocator.getInstance().repository,
             listOf(computerPlayer),
             Game.GameMode(2, Game.GameMode.Type.PC, nEvents, 0),
             artificialMovesDelay
         )
-        _gameService?.startListening()
+        gameService?.startListening()
     }
 
 
     private fun determineResult() {
-        val scores = _gameService?.currentRound!!.computeScores()
+        val scores = gameService?.currentRound!!.computeScores()
         val best = scores[0]
         val bestPoints = best.points
-        val bestUid = best.uid
-        currentUserResult = if (uid == bestUid) {
-            val secondBestPoints = scores[1].points
-            if (bestPoints > secondBestPoints) {
+        val bestScoreList = scores.filter { it.points == bestPoints }
+        currentUserResult = if (uid in bestScoreList.map { it.uid }) {
+            if (bestScoreList.size == 1) {
                 Hand.Result.WIN
             } else {
-                Hand.Result.LOSS
+                Hand.Result.TIE
             }
         } else {
-            Hand.Result.TIE
+            Hand.Result.LOSS
         }
     }
 
-    fun playHand(userHand: Hand, callback: () -> Unit) {
+    fun playHand(
+        userHand: Hand,
+        updateUIResultCallback: () -> Unit,
+        resultNavigationCallback: () -> Unit,
+        isGameOverCallback: () -> Unit
+    ) {
         viewModelScope.launch {
             //add proper round adding when supporting the multiround
             ensureActive()
-            _gameService?.playHand(userHand)
+            gameService?.playHand(userHand)
             determineResult()
-            callback()
+            updateUIResultCallback()
+            delay(1000L)
+            resultNavigationCallback()
+            delay(1000L)
```

Comment:
What are these delays for. You shouldn't add delays in the logic for no reason. Especially if it's to address a race condition.

Reply:
The delays are to let users notice the steps in the game. Otherwise, the transition would be too fast to spot.
For example, a change from the result(win/draw/loss) fragment back to the main screen has a break in between (for a user to notice the effect). In the same way, there's a break between an opponent's move and a transition to the result(win/draw/loss) page (so that they know what another player's choice was).
I hope it clarifies.

Reply:
Alright, make sure to comment to explain when you add delay, as it's very unusual so it has to be justified.

Reply:
Sure, I've added comments.